doc/release/upcoming_changes/24085.improvement.rst

```diff
@@ -0,0 +1,25 @@
+Complex types - Underlying type changes
+---------------------------------------
+
+* The underlying C types for all of numpy's complex types have been changed to
+  use C99 complex types.
```

Comment:
Is this really true across all platforms and compilers?  C99 complex types can be used with all the C arithmetic operators (+, -, *, /).  If there are still some platforms where the complex type is being defined with a struct, it won't be possible to use the arithmetic operators with it, so we shouldn't make such a broad claim here. (I just ran into this in https://github.com/numpy/numpy/pull/24416, where the attempts to use arithmetic operators fails to compile on some platforms.)

Reply:
> C99 complex types can be used with all the C arithmetic operators (+, -, *, /).

Unfortunately MSVC begs to differ. [Their documentation](https://learn.microsoft.com/en-us/cpp/c-runtime-library/complex-math-support?view=msvc-170#operation-functions) states
> Because complex numbers aren't a native type in the Microsoft compiler, the standard arithmetic operators aren't defined on complex types.

Reply:
That just means Microsoft does not implement C99--no surprise there, that has been the case forever.

Reply:
Well, the wording on that page is a bit confusing. It says that it provides all of the function required by the C99 standard, but I guess the types themselves are not C99-compatible. Should we change the wording there?